### PR TITLE
[JENKINS-38678] Remove properly administrative monitor

### PR DESCRIPTION
**Pull request by Vlatombe**
previous impl didn't work because ExtensionList#iterator() returns a
readonly iterator.

@jenkinsci/code-reviewers

### CODE REVIEW

**Comment by daniel-beck** [comment]
Should probably just deprecate `AdministrativeError` and move this to a real admin monitor that doesn't need this code. But I can understand if that's beyond what you want the scope of the PR to be.

File: core/src/main/java/jenkins/model/Jenkins.java [hunk 1]
```diff
@@ -1214,16 +1214,20 @@ private void launchTcpSlaveAgentListener() throws IOException {
...
             if (slaveAgentPort != -1 && tcpSlaveAgentListener == null) {
-                String administrativeMonitorId = getClass().getName() + ".tcpBind";
+                final String administrativeMonitorId = getClass().getName() + ".tcpBind";
                 try {
                     tcpSlaveAgentListener = new TcpSlaveAgentListener(slaveAgentPort);
                     // remove previous monitor in case of previous error
-                    for (Iterator<AdministrativeMonitor> it = AdministrativeMonitor.all().iterator(); it.hasNext(); ) {
+                    AdministrativeMonitor toBeRemoved = null;
+                    ExtensionList<AdministrativeMonitor> all = AdministrativeMonitor.all();
+                    for (Iterator<AdministrativeMonitor> it = all.iterator(); it.hasNext(); ) {
```
Comment (batmat) [inline core/src/main/java/jenkins/model/Jenkins.java]: Switch to Java 5 foreach, then, as you don't need anymore the old-school iterator style?

**Review by batmat — APPROVED** [review]
:+1:

**Comment by Vlatombe** [comment]
@daniel-beck I would prefer to minimize code change on this PR, but you're right, on the longer run, that should be replaced with a proper extension.

File: core/src/main/java/jenkins/model/Jenkins.java (hunk 1 above)
Comment (Vlatombe) [inline core/src/main/java/jenkins/model/Jenkins.java]: Good point

**Comment by recampbell** [comment]
Retriggered the build since it seems like the agent died.

**Comment by daniel-beck** [comment]
Irrelevant test failure.

:+1:

**Comment by oleg-nenashev** [comment]
:+1: IMHO it's ready to merge once we understand the status of the today's weekly

**Pull request merged**